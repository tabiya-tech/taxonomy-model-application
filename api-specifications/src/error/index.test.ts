describe("Test the error module", () => {
  test("The error module can be required via the index", async () => {
    //GIVEN the module
    //WHEN the module is required via the index
    expect(async () => await import("./")).not.toThrow(); // We check that it doesn't throw an error instead of simply letting it fail on import because we want an easier error message
    const apiErrorModule = await import("./");

    // THEN Check if Schema is defined in it
    expect(apiErrorModule.default.Schemas.Payload).toBeDefined();

    // AND check if all the constants are defined
    const constants = apiErrorModule.default.Constants;
    expect(constants).toBeDefined();
    expect(constants.ErrorCodes).toBeDefined();
    expect(constants.ReasonPhrases).toBeDefined();
  });

  test("The export module matches the snapshot", () => {
    // GIVEN the module
    // WHEN the module is required via the index
    expect(async () => {
      // THEN Check if the module can be required without error
      const apiErrorModule = await import("./");
      expect(apiErrorModule.default).toMatchSnapshot();
    }).not.toThrow();
  });
});

export {};
