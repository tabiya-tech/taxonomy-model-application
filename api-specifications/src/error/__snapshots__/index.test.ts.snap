// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the error module The export module matches the snapshot 1`] = `
{
  "Constants": {
    "ErrorCodes": {
      "BAD_REQUEST": "BAD_REQUEST",
      "INTERNAL_SERVER_ERROR": "INTERNAL_SERVER_ERROR",
      "INVALID_JSON_SCHEMA": "INVALID_JSON_SCHEMA",
      "MALFORMED_BODY": "MALFORMED_BODY",
      "METHOD_NOT_ALLOWED": "METHOD_NOT_ALLOWED",
      "NOT_FOUND": "NOT_FOUND",
      "TOO_LARGE_PAYLOAD": "TOO_LARGE_PAYLOAD",
      "UNSUPPORTED_MEDIA_TYPE": "UNSUPPORTED_MEDIA_TYPE",
    },
    "ReasonPhrases": {
      "BAD_REQUEST": "Bad Request",
      "INTERNAL_SERVER_ERROR": "Internal Server Error",
      "INVALID_JSON_SCHEMA": "Invalid json schema",
      "MALFORMED_BODY": "Payload is malformed, it should be a valid model json",
      "METHOD_NOT_ALLOWED": "Method Not Allowed",
      "NOT_FOUND": "Not Found",
      "TOO_LARGE_PAYLOAD": "Payload is too long",
      "UNSUPPORTED_MEDIA_TYPE": "Unsupported Media Type",
    },
  },
  "Schemas": {
    "Payload": {
      "$id": "/components/schemas/ErrorSchema",
      "additionalProperties": false,
      "properties": {
        "details": {
          "description": "Additional details about the error. Might be an empty string if no additional details are available",
          "type": "string",
        },
        "errorCode": {
          "description": "A code that API consumers can use to determine the type of error that occurred",
          "enum": [
            "MALFORMED_BODY",
            "METHOD_NOT_ALLOWED",
            "NOT_FOUND",
            "INTERNAL_SERVER_ERROR",
            "UNSUPPORTED_MEDIA_TYPE",
            "TOO_LARGE_PAYLOAD",
            "BAD_REQUEST",
            "INVALID_JSON_SCHEMA",
            "FAILED_TO_TRIGGER_IMPORT",
            "FAILED_TO_TRIGGER_EXPORT",
            "DB_FAILED_TO_CREATE_MODEL",
            "MODEL_COULD_NOT_VALIDATE",
          ],
          "pattern": "\\S",
          "type": "string",
        },
        "message": {
          "description": "A human readable description of the error",
          "type": "string",
        },
      },
      "required": [
        "errorCode",
        "message",
        "details",
      ],
      "type": "object",
    },
  },
  "Types": {},
}
`;
